lib/membrane_h264_plugin/parser.ex

```diff
@@ -208,19 +193,65 @@ defmodule Membrane.H264.Parser do
     {[end_of_stream: :output], state}
   end
 
-  defp prepare_actions_for_aus(aus, pts, dts, state) do
-    Enum.reduce(aus, [], fn au, acc ->
-      sps_actions =
-        case Enum.find(au, &(&1.type == :sps)) do
-          nil ->
-            []
+  defp prepare_actions_for_aus(aus, state, buffer_pts \\ nil, buffer_dts \\ nil) do
+    {actions, au_counter, profile} =
+      Enum.reduce(aus, {[], state.au_counter, state.profile}, fn au,
+                                                                 {actions_acc, cnt, profile} ->
+        {sps_actions, profile} = maybe_parse_sps(au, state, profile)
+        {pts, dts} = prepare_timestamps(buffer_pts, buffer_dts, state, profile, cnt)
 
-          sps_nalu ->
-            [stream_format: {:output, Format.from_sps(sps_nalu, framerate: state.framerate)}]
-        end
+        {actions_acc ++ sps_actions ++ [{:buffer, {:output, wrap_into_buffer(au, pts, dts)}}],
+         cnt + 1, profile}
+      end)
+
+    state = maybe_update_state(buffer_pts, buffer_dts, au_counter, profile, state)
+
+    {actions, state}
+  end
+
+  defp maybe_parse_sps(au, state, profile) do
+    case Enum.find(au, &(&1.type == :sps)) do
+      nil ->
+        {[], profile}
 
-      acc ++ sps_actions ++ [{:buffer, {:output, wrap_into_buffer(au, pts, dts)}}]
-    end)
+      sps_nalu ->
+        fmt = Format.from_sps(sps_nalu, framerate: state.framerate)
+        {[stream_format: {:output, fmt}], fmt.profile}
+    end
+  end
+
+  defp prepare_timestamps(_buffer_pts, _buffer_dts, state, profile, order_number)
```

Comment:
```suggestion
  defp prepare_timestamps(_buffer_pts, _buffer_dts, state, profile, frame_order_number)
```
How about such a name?